src/ext/java/org/opentripplanner/ext/siri/SiriTripPatternCache.java

```diff
@@ -12,31 +12,54 @@
 import javax.annotation.Nonnull;
 import org.opentripplanner.transit.model.network.StopPattern;
 import org.opentripplanner.transit.model.network.TripPattern;
-import org.opentripplanner.transit.model.network.TripPatternBuilder;
 import org.opentripplanner.transit.model.site.RegularStop;
 import org.opentripplanner.transit.model.site.StopLocation;
 import org.opentripplanner.transit.model.timetable.Trip;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 /**
- * A synchronized cache of trip patterns that are added to the graph due to GTFS-realtime messages.
+ * Threadsafe mechanism for tracking any TripPatterns added to the graph via SIRI realtime messages.
+ * This tracks only patterns added by realtime messages, not ones that already existed from the
+ * scheduled NeTEx. This is a "cache" in the sense that it will keep returning the same TripPattern
+ * when presented with the same StopPattern, so if realtime messages add many trips passing through
+ * the same sequence of stops, they will all end up on this same TripPattern.
+ * <p>
+ * Note that there are two versions of this class, this one for GTFS-RT and another for SIRI.
+ * See additional comments in the Javadoc of the GTFS-RT version of this class, whose name is
+ * simply TripPatternCache.
+ * TODO RT_AB: To the extent that double SIRI/GTFS implementations are kept, prefix all names
+ *             with GTFS or SIRI or NETEX rather than having no prefix on the GTFS versions.
  */
 public class SiriTripPatternCache {
 
-  private static final Logger log = LoggerFactory.getLogger(SiriTripPatternCache.class);
+  private static final Logger LOG = LoggerFactory.getLogger(SiriTripPatternCache.class);
 
+  // TODO RT_AB: Improve documentation. This seems to be the primary collection of added
+  //   TripPatterns, with other collections serving as indexes. Similar to TripPatternCache.cache
+  //   in the GTFS version of this class, but with service date as part of the key.
   private final Map<StopPatternServiceDateKey, TripPattern> cache = new HashMap<>();
 
+  // This appears to be a SIRI-specific index for use in GraphQL APIs.
+  // It is not present on the GTFS-RT version of this class.
   private final ListMultimap<StopLocation, TripPattern> patternsForStop = Multimaps.synchronizedListMultimap(
     ArrayListMultimap.create()
   );
 
+  // TODO RT_AB: clarify name and add documentation to this field.
   private final Map<TripServiceDateKey, TripPattern> updatedTripPatternsForTripCache = new HashMap<>();
 
+  // TODO RT_AB: generalize this so we can generate IDs for SIRI or GTFS-RT sources.
   private final SiriTripPatternIdGenerator tripPatternIdGenerator;
+
+  // TODO RT_AB: clarify name and add documentation to this field, and why it's constructor injected
```

Comment:
I believe that it's injected so that it's easier to test, because you can write a very simple mock function rather than a full blown implementation of `TransitService`.

Reply:
It looks like it's only used in one place, and I didn't spot any uses in testing. But maybe I missed something, or it's done this way to facilitate potential future testing. There may also have been a judgement that TransitService was just too broad as the SiriTripPatternCache only needed access to this one function. In which case this is a sort of anonymous sub-interface to better encapsulate the workings of TransitService.